src/index.js

```diff
@@ -1,16 +1,35 @@
 import './style.css';
+import * as allVar from './modules/variables.js';
 
-const data = [
-  { name: 'Ahmed', score: 90 },
-  { name: 'Ali', score: 80 },
-  { name: 'David', score: 99 },
-  { name: 'Alexender', score: 95 },
-];
-
-const display = () => {
+const display = (data) => {
   data.forEach((i) => {
-    document.querySelector('.infos').innerHTML += `<p class="info">${i.name} : ${i.score} </p>`;
+    allVar.List.innerHTML += `<p class="info">${i.user} : ${i.score} </p>`;
   });
 };
 
-display();
\ No newline at end of file
+fetch(allVar.BASE_URL)
+  .then((res) => res.json())
+  .then((data) => display(data?.result));
+
+allVar.Form.addEventListener('submit', async (e) => {
+  e.preventDefault();
+  const userObj = {
+    user: e.target.user.value,
+    score: e.target.score.value,
+  };
+
+  const postRequest = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(userObj),
+  };
+
+  await fetch(allVar.BASE_URL, postRequest).then((res) => res.json())
+    .then(() => window.location.reload());
```

Comment:
- [ ] You are reloading the page after a user has successfully submitted a score. This is not necessary as the user should be able to get an updated list of the current scores by clicking the 'Refresh' button.
If your intention is to get the updated scores immediately after the form is submitted, you can do it asynchronously without having to reload the entire page thereby improving UX 👍